ajax/ajaxVote.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+include('config/init.php');
+
+
+// if it gets to here, saying an upvote should happen (from the UI standpoint at least)
+//actually wait, p sure it should just like trigger it. like if it gets called on by the UI
+//		js function, then check the db, if can do, do itt
+
+
+
+
+
+
+//if a vote for this PK doesnt exist in DB, create itt
+	//then if request direction up
+		//increase vote in DB
+	//else (aka down)
+		//decrease vote in DB
+
+$exists = doesUserVoteExist($_REQUEST['eventID'], $_REQUEST['userID']);
+
+if($exists == null){
+	insertUserVote($_REQUEST['eventID'], $_REQUEST['userID'], $_REQUEST['direction']);
+	if($_REQUEST['direction'] == "up"){
+		upVoteinDB($_REQUEST['eventID']);
+	}
+	else{
+		downVoteinDB($_REQUEST['eventID']);
+	}
+}
+
+
+//if vote exists,
+
+//if downvote exists for this PK in DB, but not upVote
+	//if the variable direction = up
+		//delete the pk where direction is down
+		//create a new w pk where direction is up
+		//upvote in DB
+	//else (aka variable direction is down)
+		// then delete the vote where PK and
+		//upvote in the DB
+
+else if($exists['0']['direction'] == "down"){
```

Comment:
the logic on this page should be similar to the javascript functions I comment on below (I did those first). So I'd do that first, then come here and see if you can apply the same ideas. I left a few comments, but I can do a more complete review of this after your next round of updates.